### PR TITLE
bootutil: Add compression check function for fw validation

### PR DESCRIPTION
**Pull request by Vge0rge**
-Adds a function and explanation for the compression
 byte for the firmware validation with ECDSA

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
``` C
boot/bootutil/include/bootutil/crypto/ecdsa_p256.h:85:29: error: passing argument 1 of 'check_compression_byte' from incompatible pointer type [-Werror=incompatible-pointer-types]
  if (check_compression_byte(&pk) < 0)
                             ^
boot/bootutil/include/bootutil/crypto/ecdsa_p256.h:44:19: note: expected 'uint8_t ** {aka unsigned char **}' but argument is of type 'const uint8_t ** {aka const unsigned char **}'
 static inline int check_compression_byte(uint8_t **pk)
                   ^~~~~~~~~~~~~~~~~~~~~~
```

**Review by ioannisg — APPROVED** [review]

File: boot/bootutil/include/bootutil/crypto/ecdsa_p256.h [hunk 1]
```diff
@@ -61,10 +82,10 @@ static inline int bootutil_ecdsa_p256_verify(bootutil_ecdsa_p256_context *ctx,
     (void)sig_len;
 
     /* Only support uncompressed keys. */
-    if (pk[0] != 0x04) {
-        return -1;
-    }
-    pk++;
+	if (check_compression_byte(&pk) < 0)
+		return -1;
```
Comment (utzig) [inline boot/bootutil/include/bootutil/crypto/ecdsa_p256.h:86]: Please add brackets around the function body.

File: boot/bootutil/include/bootutil/crypto/ecdsa_p256.h [hunk 2]
```diff
@@ -38,6 +38,27 @@
 extern "C" {
 #endif
 
+#if	defined(MCUBOOT_USE_TINYCRYPT) || \
+	defined(MCUBOOT_USE_CC310) || \
+	defined(MCUBOOT_USE_NRF_EXTERNAL_CRYPTO)
+static inline int check_compression_byte(uint8_t **pk)
```
Comment (utzig) [inline boot/bootutil/include/bootutil/crypto/ecdsa_p256.h:44]: The name is a bit deceiving given that it is not just doing a check, but also changing the input.

**Review by utzig — CHANGES_REQUESTED** [review]
Please fix the issue pointed by @nvlsianpu, not sure this is gonna pass the simulator even with build fixes, but let's see.

File: boot/bootutil/include/bootutil/crypto/ecdsa_p256.h (hunk 2 above)
Comment (nvlsianpu) [reply boot/bootutil/include/bootutil/crypto/ecdsa_p256.h:44]: check_and_skip.... get_uncompresed....

**Comment by Vge0rge** [comment]
Thanks for the feedback guys! I will have to close this PR though since I face some incompatibilities with the CI downstream and here.

**Pull request closed without merging**